### ISSUE TITLE
Make jump-links more accessible

### ISSUE THREAD

**Issue by andrewlewandowski** [department-of-veterans-affairs/va.gov-team#2199]
## Problem descriptions
1. How might we provide more descriptive jump-link text in code for users relying on screen readers?
2. How might we slow down/make more elegant the jump-link interaction?
3. How might we link to accordions with unique URLs?

## Notes
- Use this page to explore solutions to the problem descriptions: https://www.va.gov/pittsburgh-health-care/locations/pittsburgh-va-medical-center-university-drive/
- Authors use Drupal to power static content pages. Please take into consideration the Drupal AX when making recommendations (e.g., If a recommendation is that we hard-code things, then how will we make sure new pages created in Drupal are compliant?)

## Objectives or Key Results this is meant to further
- A write-up of recommended solutions to the problem statements.

## Acceptance Criteria
- [ ] Make jump links slower by using `.scroll`, `.scrollTo`, or `.scrollIntoView` with the `{ behavior: 'smooth' }` option.

## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `triage`, `tools-improvements`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `call center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by 1Copenut** [comment]
@andrewlewandowski I've been researching this afternoon, and there's not a lot of information on jump links in particular. The things I've seen called out in particular were related to parallax effects and bouncing or vibrating elements on the page.

* [ALA: Accessibility for Vestibular Disorders: How My Temporary Disability Changed My Perspective](https://alistapart.com/article/accessibility-for-vestibular/)
* [ALA: Designing Safer Web Animation For Motion Sensitivity](https://alistapart.com/article/designing-safer-web-animation-for-motion-sensitivity/)

Talking with a couple of colleagues at Ad Hoc, we could consider slowing down the jump links, using our JavaScript helper method. We should make sure the links work with JS, then sprinkle the slowed down transition on as progressive enhancement.

**Comment by andrewlewandowski** [comment]
@1Copenut - Looks like there's nothing special we can do to answer question 1.

For question 2, I think it would be great if we implemented a slowed-down page scroll/jump. Can we use our JS helper for global interaction events (e.g.m page loads, accordion expand/collapse, etc.) -- like, transition-delay, ease-in, ease-out, etc.? @ncksllvn, do you have thoughts on this from a global front-end perspective?

For question 3, have you been able to do some research into this?

**Comment by andrewlewandowski** [comment]
@rtwell and @emilywaggoner - Please be thinking about global interaction design in regards to my comment about question 2 above.

**Comment by ncksllvn** [comment]
@andrewlewandowski So sorry for just now seeing this! But yes, I think we could pretty easily add the slow-scroll behavior globally by a shared JS helper. The flow would be - anchor link clicked -> event is spotted by JS -> the JS does a slow scroll via the native browser function, [scrollIntoView](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView). I would love for our team to pick this up if you think it's ready.

**Comment by andrewlewandowski** [comment]
@ncksllvn - Yes, please!

@Karak888 - Can you please add this to the Public Website backlog? Let's start with adding the JS helper to jump links, but also consider other interactions to add this to. Could @LizCallaghan help with identifying these interactions? A couple I can think of off the top of my head: new page load, and accordion open/close.

@jenniferlee-dsva - FYI, to be prioritized to pick up.

**Comment by jenstrickland** [comment]
Based on conversation with consultants at Deque and Level Access, the recommendation is that links should be instant. Smooth scroll has negative consequences for those with vestibular and cognition issues. The default behavior of going to the link is preferred. 

If our pages are very long, we may be introducing additional vestibular concerns, and it may be helpful to set a page length constraint. @1Copenut and I have been working on a recommendation for jump links in: https://github.com/department-of-veterans-affairs/va.gov-team/issues/4067 

At this time the recommendation is to not animate the transition, let it be instant.

**Comment by jenniferlee-dsva** [comment]
@1Copenut and @jenstrickland - can we jump on a quick call? Jennifer's comment above is 100% contradictory from info Andy received.

**Comment by ncksllvn** [comment]
@jenstrickland If it matters, this is an animation only when an anchor link on the same page is clicked, not the behavior of the browser reloading. There won't be an animation in the latter case.

**Comment by jenstrickland** [comment]
@ncksllvn Yes, this "smooth-scroll" for a same-page jump link is what the consultants said not to do. It may cause a reaction in someone with vestibular conditions, triggering a migraine or other illness.

**Comment by MickinSahni** [comment]
moving this out of sprint 14 and into backlog until we have a decision

**Comment by kelsonic** [comment]
@MickinSahni It sounds like this ticket can be closed as @jenstrickland noted that our current behavior is desirable.
